src/components/select/simple-select/simple-select.test.tsx

```diff
@@ -1,85 +1,794 @@
+  });
+});
+
+describe("forwarded ref", () => {
+  it("allows access to the input element through a forwarded callback ref", () => {
+    const mockRef = jest.fn((element) => element);
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} ref={mockRef}>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    expect(mockRef).toHaveBeenNthCalledWith(1, screen.getByRole("combobox"));
+  });
+
+  it("allows access to the input element through a forwarded ref object", () => {
+    const mockRef = { current: null };
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} ref={mockRef}>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    expect(mockRef.current).toBe(screen.getByRole("combobox"));
+  });
+});
+
+test("raises deprecation warning when component is used with defaultValue and no onChange (uncontrolled usage)", () => {
+  jest.spyOn(console, "warn").mockImplementation(() => {});
+
+  const loggerSpy = jest.spyOn(Logger, "deprecate");
+  render(
+    <SimpleSelect label="Colour" onChange={undefined} defaultValue="amber">
+      <Option text="amber" value="amber" />
+    </SimpleSelect>
+  );
+
+  expect(loggerSpy).toHaveBeenNthCalledWith(
+    1,
+    "Uncontrolled behaviour in `Simple Select` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
+  );
+});
+
+describe("required prop", () => {
+  it("marks input as required when required prop is true", () => {
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} required>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    expect(screen.getByRole("combobox")).toBeRequired();
+  });
+
+  it("adds asterisk to label when required prop is true", () => {
+    render(
+      <SimpleSelect label="Colour" onChange={() => {}} required>
+        <Option text="amber" value="amber" />
+      </SimpleSelect>
+    );
+
+    const label = screen.getByText("Colour");
+    expect(label).toHaveStyleRule("content", '"*"', { modifier: "::after" });
+  });
+});
+
+test("marks input as optional when isOptional prop is true", () => {
+  render(
+    <SimpleSelect label="Colour" onChange={() => {}} isOptional>
+      <Option text="amber" value="amber" />
+    </SimpleSelect>
+  );
+
+  const labelContainer = screen.getByTestId("label-container");
+  expect(labelContainer).toHaveStyleRule("content", '"(optional)"', {
+    modifier: "::after",
+  });
+});
+
+test("does not call onOpen, when openOnFocus is true and the input is refocused while the dropdown list is already open", () => {
+  jest.useFakeTimers();
+
+  const onOpen = jest.fn();
+  render(
+    <SimpleSelect
+      label="Colour"
+      onChange={() => {}}
+      onOpen={onOpen}
+      openOnFocus
+    >
+      <Option text="amber" value="amber" />
+    </SimpleSelect>
+  );
+
+  const input = screen.getByRole("combobox");
+
+  // open dropdown list
+  fireEvent.focus(input);
```

Comment:
comment (non-blocking): we can probably get away with calling `input.focus()` here as we're not testing the user interaction as part of this test etc. It feels like overkill to import `fireEvent` just for this but it's not a huge issue

Reply:
I did try this out, unfortunately it appears a condition on line 401 in `SimpleSelect`'s code is not covered when `input.focus()` is used.